mysystehis/main.tex

```diff
@@ -0,0 +1,1155 @@
+ 	 		\draw [line width=2.8pt,color=ffqqqq] (-10.3,-3.27)-- (-7.662654488417639,-0.1831069580342808);
+ 	 		\draw [line width=0.5pt] (-7.662654488417639,-0.1831069580342808)-- (-7.639990169721227,-3.2503447549486797);
+ 	 		\draw (-10.74,1.68) node[anchor=north west] {90};
+ 	 		\draw [line width=0.5pt] (1.77,-3.3) circle (2.030055417962771cm);
+ 	 		\draw [line width=0.5pt] (-2.25,-3.31)-- (5.87,-3.25);
+ 	 		\draw (-5.82,-2.98) node[anchor=north west] {0};
+ 	 		\draw (-15.86,-3.14) node[anchor=north west] {180};
+ 	 		\draw (-10.94,-7.76) node[anchor=north west] {270};
+ 	 		\draw (1.46,1.66) node[anchor=north west] {0};
+ 	 		\draw (6.42,-2.92) node[anchor=north west] {90};
+ 	 		\draw (1.28,-7.96) node[anchor=north west] {180};
+ 	 		\draw (-3.74,-3.02) node[anchor=north west] {270};
+ 	 		\draw [line width=2.8pt,color=ffqqqq] (1.77,-3.3)-- (1.7404368698694395,0.7600032045968739);
+ 	 		\draw [line width=0.5pt] (1.77,-3.3)-- (1.7995631301305606,-7.3600032045968735);
+ 	 		\begin{scriptsize}
+ 	 		\draw [fill=uuuuuu] (-10.3,-3.27) circle (2.0pt);
+ 	 		\draw[color=black] (-8.94,-3.55) node {$X$};
+ 	 		\draw[color=black] (-7.32,-1.87) node {$Y$};
+ 	 		\draw[color=qqwuqq] (-9.22,-2.87) node {$\alpha$};
+ 	 		\draw [fill=uuuuuu] (1.77,-3.3) circle (2.0pt);
+ 	 		\end{scriptsize}
+ 	 		\end{tikzpicture}
+ 	 	\\[0.02cm]
+ 	 	From the left one above, suppose that the red line is the input direction, X and Y are vertical and horizontal component. Now, i have to find the angle {$\alpha$} which is the direction the character will rotate, calculated as: 
+ 	 	\\[-0.5cm]
+ 	 	\[\alpha = arctan(\frac{Y}{X})\]
+ 	 	In Unity, however, if my character is facing forward, then it has the rotation of 0 (as seen from the right one), when facing right it has the rotation of 90 degree and so on. Let's call the rotation of character in unity is r, when looking at the two circle, it's clearly that r is calculated as: \\[-0.5cm]
+ 	 	 \[r = 90 - \alpha; \hspace{0.8cm} or \hspace{1.0cm} r = arctan(\frac{X}{Y})\] 
+ 	 	By C\# code:
+ 	 	\begin{lstlisting}
+ 	 	float targetRotation = Mathf.Atan2(inputDir.x, inputDir.y) * Mathf.Rad2Deg;
+ 	 	transform.eulerAngles = Vector3.up * Mathf.SmoothDampAngle(transform.eulerAngles.y, targetRotation, ref turnSmoothVelocity, turnSmoothTime);
+ 	 	\end{lstlisting}
+ 	 	
+ 	 	Where Mathf.SmoothDampAngle is the function that can gradually change an angle given in degrees towards a desired angle by turnSmoothTime (in second). The reason i use Vector3.up is because the character won't rotate up or fly straight to the sky.  \\[0.15cm] After that, to be able to move, transform.forward and Controller.Move is used here to move the character by a certain amount in the world space everytime when i press a button. \\[0.15cm] Finally, to finish setting up my character controller script, i have also added gravity, ability to jump for my character. From 10th grade, the equation to calculate the falling speed is defined as: 
```

Comment:
Vector3.up is used in the calculation so that the character will not rotate up or fly straight up.